src/engine/BMGame.php

```diff
@@ -2798,10 +2655,14 @@ public function __set($property, $value) {
         }
 
         $subProperty = substr($property, 0, strlen($property) - 5);
-        $isSubPropertyArray = ('Array' == substr($subProperty, -5));
+        $isDieArray = ('DieArray' == substr($subProperty, -8));
```

Comment:
This is starting to look a little magical.  Please add a comment to this block explaining what these strings are such that substringing them leads to anything good.  :>)